ledger/apply/application.go

```diff
@@ -376,6 +380,10 @@ func ApplicationCall(ac transactions.ApplicationCallTxnFields, header transactio
 		return err
 	}
 
+	if ac.RejectVersion > 0 && params.Version >= ac.RejectVersion {
```

Comment:
should this if statement be proto checked - probably not since we are feature gating the Version increment..

Reply:
My preference is to feature-gate "early" in the transaction lifecycle (WellFormed, for example) and then write the rest of my code assuming the feature exists (assuming, as is true in this case, that it degenerates you a No-op when unused). That's usually how things turn out because of the demand for backward compatibility - we have to make 0 do nothing after the upgrade because transactions will still come in without the field set and need the same behavior.